### ISSUE TITLE
code executing recursively when used with calibratedclassfier of sklearn

### ISSUE THREAD

**Issue by ravigurnatham** [catboost/catboost#1486]
model = CatBoostClassifier(depth = 10, l2_leaf_reg= 3, iterations= 500, learning_rate= 0.1, loss_function='MultiClass',thread_count=-1)
c_cfl=CalibratedClassifierCV(model,method='sigmoid')
c_cfl.fit(X_train,y_train)

so when I run the code, it executing for all the iterations from 1-500 and when it finishes it's
<img width="617" alt="Screen Shot 2020-11-12 at 1 14 18 PM" src="https://user-images.githubusercontent.com/39865757/98985323-01bbd900-24e9-11eb-9816-f56240bacec5.png">
 again starting from 1-500 and this process happening continuously and not halting. 

Is this a bug?   I attached the screenshot for reference.
